libsolidity/inlineasm/AsmCodeGen.cpp

```diff
@@ -217,42 +228,14 @@ class CodeTransform: public boost::static_visitor<>
 		int height = m_state.assembly.deposit();
 		boost::apply_visitor(*this, *_varDecl.value);
 		expectDeposit(1, height, locationOf(*_varDecl.value));
-		m_state.variables.push_back(make_pair(_varDecl.name, height));
+		solAssert(m_scope.identifiers.count(_varDecl.name), "");
```

Comment:
`identifiers` is a `std::map`, so that is ensured at the datastructure level. Here, using `count` is of course overkill, but `std::map` does not have a convenient `exists` method.